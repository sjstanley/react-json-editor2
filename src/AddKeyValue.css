.AddKeyValue {
  margin: 10px;
  padding: 10px;
  /*border: 2px solid #00FFD8;*/
  /*background: #DDFFEE;*/
  /*text-align: center;*/
}