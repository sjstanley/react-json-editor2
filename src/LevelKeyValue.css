.LevelKeyValue {
  /*border: 2px solid black;*/
  /*text-align: center;*/
  background: #f5f5f5;
  /*color: #333;*/
  /*display: none;*/
  margin-left: 10px;
  padding: 10px;
  border-left: 1px dotted #ddd;
}

.compoundAttr > .LevelKeyValue {
	display: block;
}

.objectAttr {
	float: left;
}

.attrName {
  font-style: italic;
}

.attrValue {
  margin-left: 5px;
}