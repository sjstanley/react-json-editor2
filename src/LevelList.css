.LevelList {
  margin: 10px;
  padding: 10px;
  border: 2px solid #00D8FF;
  background: #FFDDEE;
}