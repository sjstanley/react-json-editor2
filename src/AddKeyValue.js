import React, { Component } from 'react';
import './AddKeyValue.css';

class AddKeyValue extends Component {
	constructor(props) {
		super(props);
		this.state = { value: ''};
		this.handleUpdate = this.handleUpdate.bind(this);
		this.addKeyValue = this.addKeyValue.bind(this);
	}
	handleUpdate(event) {
		this.setState({ value: event.target.value });
	}
	addKeyValue() {
		this.props.addKeyValue(this.state.value);
		this.setState({ value: '' });
	}

	render() {
		return (
			<div className="AddKeyValue">
				<input 
					type="text" 
					onChange={this.handleUpdate}
					value={this.state.value}
				/>
				&nbsp;&nbsp;
				<button onClick={this.addKeyValue} >Add List Item</button>
			</div>
		);
	}
}

export default AddKeyValue;