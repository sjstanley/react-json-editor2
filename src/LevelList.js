import React, { Component } from 'react';
import './LevelList.css';
import LevelKeyValue from './LevelKeyValue';
import AddKeyValue from './AddKeyValue';

class LevelList extends Component {

	constructor(props) {
		super(props);
		this.state = {listObject: this.props.listObject};
		this.addKeyValue = this.addKeyValue.bind(this);
		this.removeKeyValue = this.removeKeyValue.bind(this);
		this.updateKeyValue = this.updateKeyValue.bind(this);
	}


	renderKeyValuePairs() {
		if (this.state.listObject) {
			return this.state.listObject.map((value, index) => (
				<LevelKeyValue 
					key={index} 
					index={index}
					value={value}
					removeKeyValue={this.removeKeyValue}
					updateKeyValue={this.updateKeyValue}
				/>
			));
		}	
	}

	addKeyValue(newValue) {
		if (newValue === '') {
			newValue = null;
		}
		this.setState({listObject: [...this.state.listObject, newValue] });
	}
	removeKeyValue(removeIndex, removeValue){
		// const filteredKeyValuePairs = this.state.listObject.filter(value => {
		// 	return value !== removeValue;});
		const filteredKeyValuePairs = this.state.listObject
		filteredKeyValuePairs.splice(removeIndex, 1);
		this.setState({ listObject: filteredKeyValuePairs });
	}
	updateKeyValue(updateIndex, changedValue) {
		const changedKeyValuePairs = this.state.listObject 
	
		changedKeyValuePairs[updateIndex] = changedValue;
		this.setState({listObject: changedKeyValuePairs});
	}
	

	render() {
		return (

			<div className="LevelList compoundAttr">
				<pre>list: </pre>
				<pre>{JSON.stringify(this.state.listObject, null, '  ')}</pre>
				{this.renderKeyValuePairs()}
				<AddKeyValue 
					addKeyValue={this.addKeyValue}
				 />
			</div>
		);
	}
}

export default LevelList;