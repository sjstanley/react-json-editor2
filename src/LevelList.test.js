import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import LevelList from './LevelList';
import AddKeyValue from './AddKeyValue';
import LevelKeyValue from './LevelKeyValue';

describe(LevelList, () => {
	const component = shallow(<LevelList />);

it('renders and matches our snapshot', () => {
	const component = renderer.create(<LevelList />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

it('contains an AddKeyValue subcomponent', () => {
	expect(component.find(AddKeyValue)).toHaveLength(1);
});

it('contains the same number of LevelKeyValue components as greetings', () => {
	const LevelKeyValues = component.find(LevelKeyValue).length;
	//const greetings = component.state.greetings.length;
	//expect(LevelKeyValues).toEqual(greetings);
	expect(LevelKeyValues).toEqual(3);



});

it('adds another greeting when the add greeting function is called', () => {
	const before = component.find(LevelKeyValue).length;
	component.instance().addGreeting('Sample');
	const after = component.find(LevelKeyValue).length;
	expect(after).toBeGreaterThan(before);
});

it('removes a greeting from the list when the remove greeting function is called', () => {
	const before = component.find(LevelKeyValue).length;
	const removeMe = component.state('greetings')[0];
	component.instance().removeGreeting(removeMe);
	const after = component.find(LevelKeyValue).length;
	expect(after).toBeLessThan(before);
});

});

