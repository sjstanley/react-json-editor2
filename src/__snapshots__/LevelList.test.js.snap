exports[`class LevelList extends /* istanbul ignore next */_react.Component {

	constructor(props) {
		super(props);
		//this.state = { greetings: ['Jim', 'Sally', 'Bender']};
		this.state = { greetings: this.props.greetings };
		this.addGreeting = this.addGreeting.bind(this);
		this.removeGreeting = this.removeGreeting.bind(this);
		this.updateGreeting = this.updateGreeting.bind(this);
	}


	renderGreetings() {
		if (this.state.greetings) {
			return this.state.greetings.map((name, index) =>
			/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_LevelKeyValue2.default, /* istanbul ignore next */{
				key: index,
				index: index,
				name: name,
				removeGreeting: this.removeGreeting,
				updateGreeting: this.updateGreeting,
				restoreOriginal: this.restoreOriginal, __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }));


		}
	}

	addGreeting(newName) {
		this.setState({ greetings: [...this.state.greetings, newName] });
	}
	removeGreeting(removeName) {
		const filteredGreetings = this.state.greetings.filter(name => {
			return name !== removeName;});

		this.setState({ greetings: filteredGreetings });
	}
	updateGreeting(originalName, changedName) {
		const changedGreetings = this.state.greetings;

		for (var k = 0; k < changedGreetings.length; k++) {
			if (originalName === changedGreetings[k]) {
				changedGreetings[k] = changedName;
			}
		}
		this.setState({ greetings: changedGreetings });
	}


	render() {
		return (

			/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'LevelList', __source: { fileName: _jsxFileName, lineNumber: 57 }, __self: this },
				/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'pre', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 58 }, __self: this }, JSON.stringify(this.state, null, '  ')),
				this.renderGreetings(),
				/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_AddKeyValue2.default, /* istanbul ignore next */{
					addGreeting: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 60 }, __self: this })));



	}} renders and matches our snapshot 1`] = `
<div
  className="LevelList">
  <pre>
    {}
  </pre>
  <div
    className="AddKeyValue">
    <input
      onChange={[Function]}
      type="text"
      value="" />
      
    <button
      onClick={[Function]}>
      Add Item
    </button>
  </div>
</div>
`;
